### DISCUSSION TITLE
How to save Cursor Object to resume later.

### DISCUSSION THREAD

**Discussion by Srj** [tweepy/tweepy#1470] [Questions] [answered]
Suppose i have a cursor object like below.

```
tweets = tw.Cursor(api.search,
                  q = search_words,
                  lang="en",
                  since=date_since,).items(1000)
```

After extracting 200 tweets i will shutdown my machine and then i want to resume this operation(means extract last 800 tweets).
What is the best way to do it?
If i pickle the cursor object ( `tweets `in the code above), will it resume the extraction process or start over again?

**Comment by Harmon758** [comment, marked as answer]
If the pagination is by ID, as is the case with [`API.search`](https://tweepy.readthedocs.io/en/latest/api.html#API.search):
https://github.com/tweepy/tweepy/blob/1a71938e81fbb829fd45bdd06d5798762078b10f/tweepy/binder.py#L257-L267  Then the iterator you're using is an `IdIterator` returned by `Cursor.pages`:
https://github.com/tweepy/tweepy/blob/1a71938e81fbb829fd45bdd06d5798762078b10f/tweepy/cursor.py#L9-L23  In which case, you can use the `max_id` attribute of the `IdIterator`:
https://github.com/tweepy/tweepy/blob/1a71938e81fbb829fd45bdd06d5798762078b10f/tweepy/cursor.py#L116-L120

Alternatively, you can just use the ID of the last Tweet you previously processed and use it for the `max_id` parameter for `API.search`.
